integration-tests/src/test/java/eu/xenit/solr/backup/s3/SolrBackupTest.java

```diff
@@ -0,0 +1,150 @@
+import static java.lang.Thread.sleep;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.restassured.RestAssured;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.parsing.Parser;
+import io.restassured.specification.RequestSpecification;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+
+public class SolrBackupTest {
+    static RequestSpecification spec;
+    static RequestSpecification specBackup;
+    static RequestSpecification specBackupDetails;
+    static RequestSpecification specRestore;
+    static RequestSpecification specRestoreStatus;
+
+    @BeforeEach
+    public void setup() {
+
+        String basePathSolr = "solr/alfresco";
+        String basePathSolrBackup = "solr/alfresco/replication";
+        String solrHost = System.getProperty("solr.host","localhost");
+        int solrPort = 0;
+        try {
+            solrPort = Integer.parseInt(System.getProperty("solr.tcp.8080","8080"));
+        } catch(NumberFormatException e) {
+            System.out.println("Solr port 8080 is not exposed, probably ssl is enabled");
+        }
+
+        System.out.println("basePathSolr=" + basePathSolr + " and basePathSolrBackup=" + basePathSolrBackup + " and solrHost=" + solrHost + " and solrPort=" + solrPort);
+        String baseURISolr = "http://" + solrHost;
+
+        RestAssured.defaultParser = Parser.JSON;
+
+        spec = new RequestSpecBuilder()
+                .setBaseUri(baseURISolr)
+                .setPort(solrPort)
+                .setBasePath(basePathSolr)
+                .build();
+        specBackup = new RequestSpecBuilder()
+                .setBaseUri(baseURISolr)
+                .setPort(solrPort)
+                .setBasePath(basePathSolrBackup)
+                .addParam("command","backup")
+                .addParam("repository","s3")
+                .addParam("location","s3:///")
+                .addParam("numberToKeep","3")
+                .build();
+        specBackupDetails = new RequestSpecBuilder()
+                .setBaseUri(baseURISolr)
+                .setPort(solrPort)
+                .setBasePath(basePathSolrBackup)
+                .addParam("command","details")
+                .addParam("wt","json")
+                .build();
+        specRestore = new RequestSpecBuilder()
+                .setBaseUri(baseURISolr)
+                .setPort(solrPort)
+                .setBasePath(basePathSolrBackup)
+                .addParam("command","restore")
+                .addParam("repository","s3")
+                .addParam("location","s3:///")
+                .build();
+        specRestoreStatus = new RequestSpecBuilder()
+                .setBaseUri(baseURISolr)
+                .setPort(solrPort)
+                .setBasePath(basePathSolrBackup)
+                .addParam("command","restorestatus")
+                .addParam("wt","json")
+                .build();
+
+
+        // wait for solr to track
```

Comment:
is there a solr REST API to query the tracking status/progress ?
together with `awaitility` this could be a lot more elegant